Add logger

mohammadranjbarz opened the pull request:
related to #302

mohammadranjbarz committed:
Add bunyan logger

mohammadranjbarz committed:
Add log path and bind log path in docker-compose

mohammadranjbarz committed:
Merge branch 'staging' into add-logger

mohammadranjbarz committed:
erge staging to add-logger

mohammadranjbarz committed:
Run tslint in CI/CD

mohammadranjbarz committed:
Add huskey hook for pre-commit

mohammadranjbarz committed:
Replace all console.log with logger.debug

mohammadranjbarz committed:
Merge staging to add-logger

mohammadranjbarz committed:
Update README for see logs

CarlosQ96 commented on src/middleware/logger.ts:
```diff
@@ -1,9 +1,9 @@
+//   logger.debug('args: ', args);
+//
+//   return next();
+// };
```
If this file is not gonna be used, let's delete it.

CarlosQ96 commented on src/delete-uniswap/sdk.ts:
```diff
@@ -52,8 +53,8 @@ export default class Sdk {
-    // console.log('inv', route.midPrice.invert().toSignificant(6)) // 0.00496756
-    // console.log('price', price) // 201.306
+    // logger.debug('inv', route.midPrice.invert().toSignificant(6)) // 0.00496756
+    // logger.debug('price', price) // 201.306
```
I assume this comments serve as documentation? but they will break if we try to use them. I'll add the dependency.

mohammadranjbarz replied:
Thanks man

CarlosQ96 commented on src/config.ts:
```diff
@@ -91,7 +92,7 @@ class Config {
       if (envFile[envVar]) {
         this[envVar] = envFile[envVar];
-        // console.log(`envVar ---> : ${this[envVar]}`)
+        // logger.debug(`envVar ---> : ${this[envVar]}`)
```
I'll add the logger dependency.

CarlosQ96 commented on src/services/syncGivPrices.ts:
```diff
@@ -8,7 +8,7 @@ const cronJobTime =
 
 export const runUpdateHistoricGivPrices = () => {
-  console.log('runUpdateHistoricGivPrices() has been called');
+  logger.debug('runUpdateHistoricGivPrices() has been called');
```
This line broke we missed the import. I'll add it.

CarlosQ96 commented on src/server/components/FilterListedComponent.jsx:
```diff
@@ -5,7 +5,7 @@ import { FormGroup, filterStyles, Label } from '@admin-bro/design-system'
 const FilterListedComponent = props => {
   const { onChange, property, theme } = props;
-  console.log(props);
+  logger.debug(props);
```
Removing this, I added it for debugging purposes but it's not required. This is a adminBro custom component in jsx.

CarlosQ96 committed:
add missing logger imports

CarlosQ96 committed:
remove old logger

CarlosQ96 reviewed (approved):
Checked every single file didn't miss the import.
I added the missing dependencies on some files so they don't break and removed the old logger. Ran the server and a few processes and they seem to work fine.
Great that we set a rotation for the logger I think it's fine. Good job!
We can merge tomorrow.
LGTM!!

mohammadranjbarz commented:
> Checked every single file didn't miss the import. I added the missing dependencies on some files so they don't break and removed the old logger. Ran the server and a few processes and they seem to work fine. Great that we set a rotation for the logger I think it's fine. Good job! We can merge tomorrow. LGTM!!

Thanks for your review and fixing errors

mohammadranjbarz committed:
Resolve merge conflicts
